src/index.js

```diff
@@ -4,9 +4,9 @@ import syntax from 'babel-plugin-syntax-dynamic-import';
 const TYPE_IMPORT = 'Import';
 
 const buildImport = template(`
-  (new Promise(resolve => {
-    resolve(require(SOURCE));
-  }))
+  Promise.resolve().then(() => {
```

Comment:
this could even be a one liner, fwiw - `Promise.resolve().then(() => require(SOURCE))`?